packages/sanity/src/core/form/components/formField/FormFieldBaseHeader.tsx

```diff
@@ -47,17 +56,66 @@ const SlotBox = styled(Box)<{
 })
 
 const FieldActionsFloatingCard = styled(Card)(({theme}: {theme: Theme}) => {
-  const {space} = theme.sanity
+  const space = theme.sanity.space[1] / 2
+
   return css`
     align-items: center;
     bottom: 0;
-    gap: ${space[1] / 2}px;
-    padding: ${space[1] / 2}px;
+    gap: ${space}px;
+    padding: ${space}px;
     position: absolute;
     right: 0;
+
+    @media (hover: hover) {
+      // If hover is supported, we hide the floating card by default
+      // and only show it when it has focus within or when the field is hovered or focused.
+      opacity: 0;
+
+      [data-ui='FieldActionFlex'] {
+        opacity: 0;
+      }
+
+      &[data-actions-visible='false']:not(:focus-within) {
+        // Remove the shadow when the field actions are not visible
+        box-shadow: none;
+
+        // Since the field actions always will be present in the DOM (to make them focusable) –
+        // they will always affect the width of the floating card, even when they are not visible.
+        // Therefore, we remove the background of the floating card when the field actions are not visible.
+        background: transparent;
```

Comment:
I think we need to set `width: 0` as well – otherwise presence will be position to much to the left